burn-dataset/src/source/huggingface/importer.py

```diff
@@ -0,0 +1,175 @@
+
+    print("*"*80)
+    print("Starting huggingface dataset download and export")
+    print(f"Dataset Name: {name}")
+    print(f"Subset Name: {subset}")
+    print(f"Sqlite database file: {db_file}")
+    if cache_dir is None:
+        print(f"Custom cache dir: {cache_dir}")
+    print("*"*80)
+
+    # Load the dataset
+    dataset_all = load_dataset(
+        name, subset, cache_dir=cache_dir, use_auth_token=token)
+
+    print(f"Dataset: {dataset_all}")
+
+    # Create the database connection descriptor (sqlite)
+    engine = create_engine(f"sqlite:///{db_file}")
+
+    # Set some sqlite pragmas to speed up the database
+    event.listen(engine, 'connect', set_sqlite_pragma)
+
+    # Add an row_id column to each table as primary key (datasets does not have API for this)
+    event.listen(Table, 'before_create', add_pk_column)
+
+    # Export each split in the dataset
+    for key in dataset_all.keys():
+        dataset = dataset_all[key]
+
+        # Disable decoding for audio and image fields
+        dataset = disable_decoding(dataset)
+
+        # Flatten the dataset
+        dataset = dataset.flatten()
+
+        # Rename columns to remove dots from the names
+        dataset = rename_colums(dataset)
+
+        print(f"Saving dataset: {name} - {key}")
+        print(f"Dataset features: {dataset.features}")
+
+        # Save the dataset to a sqlite database
+        dataset.to_sql(
+            key,  # table name
+            engine,
+            # don't save the index, use row_id instead (index is not unique)
+            index=False,
+            dtype=blob_columns(dataset),   # save binary columns as blob
+        )
+
+    # Print the schema of the database so we can reference the columns in the rust code
+    print_table_info(engine)
+
+
+def disable_decoding(dataset):
+    """
+    Disable decoding for audio and image fields. The fields will be saved as raw file bytes.
+    """
+    for k, v in dataset.features.items():
+        if isinstance(v, Audio):
+            dataset = dataset.cast_column(k, Audio(decode=False))
+        elif isinstance(v, Image):
+            dataset = dataset.cast_column(k, Image(decode=False))
+
+    return dataset
+
+
+def rename_colums(dataset):
+    """
+    Rename columns to remove dots from the names. Dots appear in the column names because of the flattening.
+    Dots are not allowed in column names in rust and sql (unless quoted). So we replace them with underscores.
+    This way there is an easy name mapping between the rust and sql columns.
+    """
+
+    for name in dataset.features.keys():
+        if '.' in name:
+            dataset = dataset.rename_column(name, name.replace('.', '_'))
+
+    return dataset
+
+
+def blob_columns(dataset):
+    """
+    Make sure all binary columns are blob columns in the database
+    """
+    type_mapping = {}
+    for name, value in dataset.features.items():
+        if pa.types.is_binary(value.pa_type):
+            type_mapping[name] = LargeBinary
```

Comment:
If `pa.types` is not a binary, should we throw an error here?

Reply:
Actually, we want to make sure to use BLOB sql type for binary values. For some reason, to_sql exports as TEXT. I can document this strange behavior.